nilearn/plotting/html_stat_map.py

```diff
@@ -1,157 +1,496 @@
-                       'in future without warning. \nPlease do not rely on it '
-                       'in critical and production environments.'
-                       )
-    warnings.warn(message=change_warn_msg, category=FutureWarning)
-    
-    stat_map_img = check_niimg_3d(stat_map_img, dtype='auto')
-    if bg_img == 'MNI152':
-        bg_img = datasets.load_mni152_template()
-    if bg_img is not None and bg_img is not False:
-        bg_img = image.load_img(bg_img)
-        stat_map_img = image.resample_to_img(stat_map_img, bg_img)
-        bg_mask = image.new_img_like(bg_img, bg_img.get_data() != 0)
-        stat_map_img = image.new_img_like(
-            stat_map_img, stat_map_img.get_data() * bg_mask.get_data())
-        encoded_bg = '"{}"'.format(_encode_nii(bg_img))
-    else:
-        encoded_bg = 'null'
-    colors = colorscale(
-        cmap, stat_map_img.get_data().ravel(),
-        threshold=threshold, symmetric_cmap=symmetric_cmap, vmax=vmax)
-    abs_threshold = colors['abs_threshold']
-    abs_threshold = 0. if abs_threshold is None else float(abs_threshold)
-    papaya_cmaps = _to_papaya(
-        colors['cmap'], colors['norm'],
-        abs_threshold, colors['vmax'], colors['symmetric_cmap'])
-    stat_map_params = {
-        'min': abs_threshold, 'max': float(colors['vmax']),
-        'cmap': papaya_cmaps, 'symmetric': colors['symmetric_cmap']}
-    html = get_html_template('stat_map_template.html')
-    html = html.replace(
-        'INSERT_STAT_MAP_DATA_HERE', '"{}"'.format(_encode_nii(stat_map_img)))
-    html = html.replace('INSERT_MNI_DATA_HERE', encoded_bg)
-    html = html.replace('INSERT_STAT_MAP_PARAMS_HERE',
-                        json.dumps(stat_map_params))
-    return StatMapView(html)
+    html : the html viewer object.
+        It can be saved as an html page `html.save_as_html('test.html')`,
+        or opened in a browser `html.open_in_browser()`.
+        If the output is not requested and the current environment is a Jupyter
+        notebook, the viewer will be inserted in the notebook.
+    """
+
+    # Start by fixing the cmap for duplicated colors
+    cmap = _sanitize_cm(cmap)
+
+    # Mask stat map
+    mask_img, stat_map_img, data, threshold = _mask_stat_map(
+        stat_map_img, threshold)
+
+    # Get color bar and data ranges
+    colors = colorscale(cmap, data.ravel(), threshold=threshold,
+                        symmetric_cmap=symmetric_cmap, vmax=vmax)
+
+    # Load background image
+    bg_img, bg_min, bg_max, black_bg = _load_bg_img(stat_map_img, bg_img,
+                                                    black_bg, dim)
+
+    # Resample stat map
+    stat_map_img, mask_img = _resample_stat_map(stat_map_img, bg_img, mask_img,
+                                                resampling_interpolation)
+
+    # Get the slices for the cut
+    cut_slices = _get_cut_slices(stat_map_img, cut_coords, threshold)
+
+    # Now that we preped the data, encode the data in a sprite object
```

Comment:
Unnecessary comment

Reply:
Removed most comments in that function. Function names are all self-explanatory now.